### PR TITLE
refactor(api): Implement an API v1 Executor

### LINKED ISSUE #3538: Apiv2 Backcompat: Implement an apiv1 executor
#3537 introduced the capability to tell if a protocol is apiv1 or apiv2 and give it to the correct executor. This ticket specifies the executor.

The executor should prepare an environment with the appropriate singletons (as defined in #3534) and execute the apiv1 protocol. It should ideally do this inline in a function to prevent users from accessing the shim layer outside of protocol context.

This should also establish a command-line executable entrypoint for running a protocol. This should also work in simulation.

### PR DESCRIPTION
**Pull request by Laura-Danielle**
## overview

Closes #3538. With this PR, you can now _upload_ an api v1 protocol with the api v2 feature flag flipped. *Note* The protocol will not run correctly as the rest of the tickets in the [backcompat milestone](https://github.com/Opentrons/opentrons/milestone/103).

## review requests

Please run some protocols in different use-cases including through the app/jupyter/commandline in v1 and v2.

### CODE REVIEW

**Comment by sfoster1** [comment]
## Testing Tasks

With the APIv2 feature flag set through the runapp:
- Upload and run an apiv1 python protocol and make sure there are no exceptions
- Upload and run a json protocol and make sure there are no exceptions
- Run a python protocol from the robot command line by executing it directly (`python my_protocol.py`) and make sure there are no exceptions
- Simulate a python and json protocol with `opentrons_simulate` and make sure there are no exceptions
- Run a python protocol from the robot command line with `opentrons_execute` (`opentrons_execute my_protocol.py`) and make sure there are no exceptions
- Run a json protocol from the robot command line with `opentrons_execute` (`opentrons_execute my_protocol.py`) and make sure there are no exceptions
- Use jupyter to `from opentrons import robot, labware, instruments, modules` and run a couple basic commands and make sure there are no exceptions
- Run deck calibration from the robot command line and make sure there are no exceptions
- Flash a pipette from the robot command line `python -m opentrons.tools.write_pipette_memory`) and make sure there are no exceptions
- Run the factory tests from the command line (`python -m opentrons.tools.gantry_test`, `python -m opentrons.tools.z_stage_test`, `python -m opentrons.tools.factory_test`, `python -m opentrons.tools.overnight_test` (just let this one run through a few cycles)) and make sure there are no exceptions

On APIv2:
## Testing Tasks

With the APIv2 feature flag set through the runapp:
- Upload and run an apiv2 python protocol and make sure there are no exceptions
- Upload an apiv1 protocol and see that the exception is about missing commands or malformed commands, _not_ that the protocol is of the wrong kind
- Simulate an apiv2 python and json protocol and make sure there are no exceptions
- Simulate an apiv1 python protocol and make sure that there are only exceptions about missing commands or malformed commands, _not_ that the protocol is of the wrong kind
- Upload and run a json protocol and make sure there are no exceptions
- Run a python protocol from the robot command line with `opentrons_execute` (`opentrons_execute my_protocol.py`) and make sure there are no exceptions
- Run a json protocol from the robot command line with `opentrons_execute` (`opentrons_execute my_protocol.py`) and make sure there are no exceptions
- Use jupyter to `from opentrons import execute; api = execute.get_protocol_api()` and run a couple basic commands and make sure there are no exceptions
- Run deck calibration from the robot command line and make sure there are no exceptions
- Flash a pipette from the robot command line `python -m opentrons.tools.write_pipette_memory`) and make sure there are no exceptions
- Run the factory tests from the command line (`python -m opentrons.tools.gantry_test`, `python -m opentrons.tools.z_stage_test`, `python -m opentrons.tools.factory_test`, `python -m opentrons.tools.overnight_test` (just let this one run through a few cycles)) and make sure there are no exceptions

File: api/src/opentrons/simulate.py [hunk 1]
```diff
@@ -15,14 +15,14 @@
...
-import opentrons.legacy_api.protocols
 import opentrons.commands
 import opentrons.broker
+import opentrons.config
 from opentrons.protocols import parse, bundle
-from opentrons.protocols.types import (
-    JsonProtocol, PythonProtocol, BundleContents)
 from opentrons.protocol_api import execute
 from .util.entrypoint_util import labware_from_paths, datafiles_from_paths
+from opentrons.protocols.types import (
```
Comment (btmorr) [inline api/src/opentrons/simulate.py]: [nitpick] Unnecessary change to diff

File: api/tests/opentrons/server/calibration_integration_test.py [hunk 2]
```diff
@@ -75,6 +76,8 @@ def dummy_read_model(mount):
     else:
         expected1 = pt1
 
+    print(absolute(hardware.poses, pipette))
```
Comment (btmorr) [inline api/tests/opentrons/server/calibration_integration_test.py]: Is this add intentional?

File: api/src/opentrons/data_storage/database.py [hunk 3]
```diff
@@ -147,68 +145,63 @@ def _get_db_version(db):
 
 # ======================== Public Functions ======================== #
 def save_new_container(container: Container, container_name: str) -> bool:
-    db_conn = sqlite3.connect(database_path)
+    db_conn = sqlite3.connect(str(CONFIG['labware_database_file']))
```
[the remark this answers is not in the document]
Comment (sfoster1) [reply api/src/opentrons/data_storage/database.py]: Yeah unfortunately this was the minimal change. Other modules check this config variable and it's possible for it to get out of sync with database_path

File: api/src/opentrons/drivers/smoothie_drivers/__init__.py [hunk 4]
```diff
@@ -21,6 +21,21 @@ def home(self, axis):
...
+    def read_pipette_model(self, mount):
+        pass
+
+    def write_pipette_id(self, mount, id):
+        pass
+
+    def write_pipette_model(self, mount, model):
+        pass
+
+    def _send_command(self, command, timeout=None):
```
[the remark this answers is not in the document]
Comment (sfoster1) [reply api/src/opentrons/drivers/smoothie_drivers/__init__.py]: Yeah this is about scratching together compatiblity for factory scripts, unfortunately

File: api/src/opentrons/deck_calibration/endpoints.py [hunk 5]
```diff
@@ -132,7 +136,8 @@ def get_pipettes(sess):
         left = attached_pipettes.get('left')
         right = attached_pipettes.get('right')
         if left['model'] in pipette_config.config_models:
-            left_pipette = instruments.pipette_by_name('left', left['name'])
+            left_pipette = instruments.pipette_by_name(
```
[the remark this answers is not in the document]
Comment (btmorr) [reply api/src/opentrons/deck_calibration/endpoints.py]: Ah, protected by v2 flag--wasn't easy to see
